Specs/Scene/ModelExperimental/InstancingPipelineStageSpec.js

```diff
@@ -0,0 +1,238 @@
+      gltfResource: resource,
+      incrementallyLoadTexture: false,
+    });
+  }
+
+  function loadGltf(gltfPath, options) {
+    var gltfLoader = new GltfLoader(getOptions(gltfPath, options));
+    gltfLoaders.push(gltfLoader);
+    gltfLoader.load();
+
+    return waitForLoaderProcess(gltfLoader, scene);
+  }
+
+  it("correctly computes instancing TRANSLATION min and max from typed arrays", function () {
+    var renderResources = {
+      attributeIndex: 0,
+      attributes: [],
+      instancingTranslationMax: undefined,
+      instancingTranslationMin: undefined,
+      shaderBuilder: new ShaderBuilder(),
+    };
+
+    return loadGltf(boxInstanced).then(function (gltfLoader) {
+      var components = gltfLoader.components;
+      var node = components.nodes[0];
+
+      scene.renderForSpecs();
+      InstancingPipelineStage.process(renderResources, node, scene.frameState);
+
+      expect(renderResources.instancingTranslationMax).toEqual(
+        new Cartesian3(2, 2, 0)
+      );
+      expect(renderResources.instancingTranslationMin).toEqual(
+        new Cartesian3(-2, -2, 0)
+      );
+      expect(renderResources.attributes.length).toBe(3);
+    });
+  });
+
+  it("sets instancing TRANSLATION min and max from attributes", function () {
+    var renderResources = {
+      attributeIndex: 0,
+      attributes: [],
+      instancingTranslationMax: undefined,
+      instancingTranslationMin: undefined,
+      shaderBuilder: new ShaderBuilder(),
+    };
+
+    return loadGltf(boxInstancedTranslationMinMax).then(function (gltfLoader) {
+      var components = gltfLoader.components;
+      var node = components.nodes[0];
+
+      InstancingPipelineStage.process(renderResources, node);
+
+      expect(renderResources.instancingTranslationMax).toEqual(
+        new Cartesian3(2, 2, 0)
+      );
+      expect(renderResources.instancingTranslationMin).toEqual(
+        new Cartesian3(-2, -2, 0)
+      );
+      expect(renderResources.attributes.length).toBe(1);
+    });
+  });
+
+  it("creates instancing matrices vertex attributes when ROTATION is present", function () {
+    var renderResources = {
+      attributeIndex: 0,
+      attributes: [],
+      instancingTranslationMax: undefined,
+      instancingTranslationMin: undefined,
+      shaderBuilder: new ShaderBuilder(),
+    };
+
+    return loadGltf(boxInstanced).then(function (gltfLoader) {
+      var components = gltfLoader.components;
+      var node = components.nodes[0];
+
+      scene.renderForSpecs();
+      InstancingPipelineStage.process(renderResources, node, scene.frameState);
+
+      expect(renderResources.instancingTranslationMax).toEqual(
+        new Cartesian3(2, 2, 0)
+      );
+      expect(renderResources.instancingTranslationMin).toEqual(
+        new Cartesian3(-2, -2, 0)
+      );
+      expect(renderResources.attributes.length).toBe(3);
+
+      var attributeLines = renderResources.shaderBuilder._attributeLines;
+      expect(attributeLines[0]).toEqual(
```

Comment:
I notice the tests below check the `#define` lines. Should this test do the same or was this intentionally omitted?

Reply:
This test should also have them.